### PR TITLE
#45: Parsing table in yaml file has its issue

### LINKED ISSUE #45: Parsing table in yaml file has its issue
## Describe the bug
Certain setting are not working well in the current implementation of yaml parsing.
I also requires the boolean to be text.
Finally the field names case is inconsistent

## Expected behavior
Parsing is smooth and consistent

### PR DESCRIPTION
**Pull request by benedeki**
* `DBTableFromYaml` changed to be a full-case classes with further sub-classes, no Maps
* Add Circe configuration to convert snake case to camel case and to use default values for missing keys
* `DBTableFromYaml` now has a constructor to be create from (yaml) string
* package `helpers` renamed to `yaml`
* fixed example `my_table.yaml`
* both *.yml  and *.yaml are recognized as table source files
* Tests of `PGTableParser` and `DBTableFromYaml`

Closes #45

### CODE REVIEW

File: src/main/scala/za/co/absa/ultet/util/parsers/yaml/ColumnFromYaml.scala [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+import io.circe.generic.extras.{Configuration, ConfiguredJsonCodec}
+import za.co.absa.ultet.model.table.column.TableColumn
+import za.co.absa.ultet.types.table.ColumnName
+
+@ConfiguredJsonCodec
+case class ColumnFromYaml(
+                            columnName: String,
+                            dataType: String,
+                            notNull: Boolean,
```
Comment (lsulak) [inline src/main/scala/za/co/absa/ultet/util/parsers/yaml/ColumnFromYaml.scala:27]: will be always present?

File: src/main/scala/za/co/absa/ultet/util/parsers/yaml/DBTableFromYaml.scala [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import za.co.absa.ultet.types.DatabaseName
+import za.co.absa.ultet.types.schema.SchemaName
+import za.co.absa.ultet.types.table.{TableIdentifier, TableName}
+import za.co.absa.ultet.types.user.UserName
+
+@ConfiguredJsonCodec
+case class DBTableFromYaml(
+  table: String,
+  description: Option[String],
+  primaryDb: String,
```
Comment (lsulak) [inline src/main/scala/za/co/absa/ultet/util/parsers/yaml/DBTableFromYaml.scala:33]: why `primary` ?

also, isn't schema missing?

File: src/test/scala/za/co/absa/ultet/util/parsers/yaml/DBTableFromYamlTest.scala [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+package za.co.absa.ultet.util.parsers.yaml
+
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+import za.co.absa.ultet.types.schema.SchemaName
+
+class DBTableFromYamlTest extends AnyFlatSpec with Matchers {
+
+  private val schemaName = SchemaName("testSchema")
```
Comment (lsulak) [inline src/test/scala/za/co/absa/ultet/util/parsers/yaml/DBTableFromYamlTest.scala:25]: not used

File: src/test/scala/za/co/absa/ultet/util/parsers/PgTableFileParserTest.scala [hunk 4]
```diff
@@ -25,72 +25,29 @@ import za.co.absa.ultet.types.DatabaseName
 import za.co.absa.ultet.types.schema.SchemaName
 import za.co.absa.ultet.types.table.{ColumnName, IndexName, TableIdentifier, TableName}
 import za.co.absa.ultet.types.user.UserName
-import za.co.absa.ultet.util.parsers.helpers.DBTableFromYaml
+import za.co.absa.ultet.util.parsers.yaml.DBTableFromYaml
```
Comment (lsulak) [inline src/test/scala/za/co/absa/ultet/util/parsers/PgTableFileParserTest.scala:28]: DBTableFromYaml not used

File: src/main/scala/za/co/absa/ultet/util/parsers/yaml/SecondaryIndexFromYaml.scala [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.ultet.util.parsers.yaml
+
+import io.circe.generic.extras.{Configuration, ConfiguredJsonCodec}
+import za.co.absa.ultet.model.table.index.TableIndex.{IndexColumn, SecondaryIndex}
```
Comment (lsulak) [inline src/main/scala/za/co/absa/ultet/util/parsers/yaml/SecondaryIndexFromYaml.scala:20]: Configuration and IndexColumn not used

File: src/main/scala/za/co/absa/ultet/util/parsers/yaml/PrimaryKeyFromYaml.scala [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.ultet.util.parsers.yaml
+
+import io.circe.generic.extras.{Configuration, ConfiguredJsonCodec}
```
Comment (lsulak) [inline src/main/scala/za/co/absa/ultet/util/parsers/yaml/PrimaryKeyFromYaml.scala:19]: Configuration not used

**Review by lsulak — COMMENTED** [review]
Review finished, I didn't find any conceptual problems. Good to see Circe in a good use like this.


- [x]  code reviewed
- [x]  pulled
- [x]  built
- [x]  ran tests

Not gonna approve just yet because of some tiny issues, but otherwise it's there soon

File: src/main/scala/za/co/absa/ultet/util/parsers/yaml/DBTableFromYaml.scala (hunk 2 above)
Comment (benedeki) [reply src/main/scala/za/co/absa/ultet/util/parsers/yaml/DBTableFromYaml.scala:33]: Nope. Schema is decided by the folder the source is in - so much easier to mover if needed.

Comment (benedeki) [reply src/main/scala/za/co/absa/ultet/util/parsers/yaml/DBTableFromYaml.scala:33]: > why `primary` ?

For eventual multi-db (cluster) support. I know it's far ahead, but it's a remnant from the original design.

File: src/main/scala/za/co/absa/ultet/util/parsers/yaml/ColumnFromYaml.scala (hunk 1 above)
Comment (benedeki) [reply src/main/scala/za/co/absa/ultet/util/parsers/yaml/ColumnFromYaml.scala:27]: The default can be added later on easily. So will keep it this way for now.

**Comment by benedeki** [comment]
Suggestions applied, questions answered. Merging as it's still in pre-release regime.

**Pull request merged**